### ISSUE TITLE
BiglyBT stops listening on TCP listen port (UDP listening is fine)

### ISSUE THREAD

**Issue by david-bakin** [BiglySoftware/BiglyBT#942]
(Windows Server 2016; BiglyBt Beta 1.8.0.1_B34.)

I'm having troubles with incoming TCP connections.  I've checked and rechecked my setup and ports should be forwarded correctly.  But frequently I can _not_ get incoming TCP on the chosen port (53151) even though I _do_ get incoming UDP on the chosen port (53151).

The attached screenshot shows that both the TCP and UDP listen port are set to 53151; down at the bottom left you see that there are in fact connections to external machines on TCP 53151 - but BiglyBt is _not listening_ for TCP on 53151, only for UDP on 53151.  (According to wininternals Process Explorer.)

This BiglyBt process is only a couple of minutes old - when I started it and started a torrent _it was listening for both TCP 53151 and UDP 53151.  But after a little bit the TCP 53151 listener vanished - and it does _not_ come back.

The network check shows:

```
Testing HTTP outbound
    Test successful
Testing TCP outbound
    Test successful
Testing UDP outbound
    Test successful
Testing TCP port 53151 inbound
    Test failed: NAT test failed: Connect attempt to 24.16.204.95:53151 (your computer) timed out after 20 seconds. This means your port is probably closed
    Check your port forwarding for TCP 53151
Testing UDP port 53151 inbound
    Sending outbound packet and waiting for reply probe (timeout=5000)
    Sending completion event
    Test successful
```

![BiglyBt-not-listening-on-TCP-53151](https://user-images.githubusercontent.com/4162948/54659729-467f3a80-4a90-11e9-95f6-2d53d851f356.png)

**Comment by parg** [comment]
hmm, do you get a warning triangle bottom right of BiglyBT that when clicked on shows an issue with the TCP listen? BiglyBT periodically connects to itself to try and verify that the listener is established and will try to 'repair things' if it doesn't work.

**Comment by res1492** [comment]
The same thing happened on my pc....i spent ages thinking it was something i had changed...after 2 hours of messing about i reverted to the stable version and as was fine again...at the time i didn't think it was a bug so didn't report it but i had the same issue with listening on TCP listen port, using port checker showed that it was listening on UDP but not on TCP

**Comment by fe31nz** [comment]
I have just installed BiglyBT 1.9.0.0 today on my Windows 7 PC and I am seeing this problem too.  It is an install with the config copied from my Azuerus setup, with the open TCP and UDP ports on 17273.  Immediately after I started BiglyBT, I did Tools/NAT Firewall test, and that showed that both ports were open.  Then I ran Plugins/Network Status, and that showed UDP outbound and TCP inbound failures, and also "No incoming connections received, likely NAT problems" from the IPv6 testing.  So I tried running Tools/NAT Firewall Test again and it now showed a NAT error when doing the "Incoming TCP Listen Port" test.  When I ran Wireshark, I found that I could see inbound TCP SYN packets on port 17273, but they were all getting RST replies.  So I then ran netstat -a -b to see what ports BiglyBT was listening on, and found that it was not listening on TCP port 17273 in either IPv4 or IPv6.

I did get an error triangle on startup - the error message was saying it was unable to install the DNS manager.

This PC has two Ethernet cards, one connecting to my "Outer" network and on to the Internet, and one connecting to my "Inner" network, so I have BiglyBT set up to only bind to the static IPv4 and IPv6 addresses on the "Outer" Ethernet port.

Everything seems to be working properly on UDP port 17273.
